src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskExecuteReconfigureControllerCFTTest.java

```diff
@@ -0,0 +1,91 @@
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.entities.TaskFilter;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.entities.TaskOperation;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskFilterOperator;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskOperationName;
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.TestAuthenticationCredentials;
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.TestVariables;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.UUID;
+
+public class PostTaskExecuteReconfigureControllerCFTTest extends SpringBootFunctionalBaseTest {
+
+    private static final String ENDPOINT_BEING_TESTED = "/task/operation";
+    private TestAuthenticationCredentials caseworkerCredentials;
+    private String assigneeId;
+
+    @Before
+    public void setUp() {
+        caseworkerCredentials = authorizationProvider.getNewTribunalCaseworker("wa-ft-test-r2-");
+        assigneeId = getAssigneeId(caseworkerCredentials.getHeaders());
+    }
+
+    @After
+    public void cleanUp() {
+        common.clearAllRoleAssignments(caseworkerCredentials.getHeaders());
+        authorizationProvider.deleteAccount(caseworkerCredentials.getAccount().getUsername());
+    }
+
+    @Test
+    public void should_return_a_204_after_tasks_are_executed_for_reconfigure() {
+        TestVariables taskVariables = common.setupTaskAndRetrieveIds();
+        common.setupCFTOrganisationalRoleAssignment(caseworkerCredentials.getHeaders(), "IA", "Asylum");
+        initiateTask(caseworkerCredentials.getHeaders(), taskVariables,
+            "followUpOverdueReasonsForAppeal", "follow Up Overdue Reasons For Appeal", "A test task");
+
+        String assignEndpoint = "task/{task-id}/assign";
+        String taskId = taskVariables.getTaskId();
+        Response result = restApiActions.post(
+            assignEndpoint,
+            taskId,
+            new AssignTaskRequest(assigneeId),
+            caseworkerCredentials.getHeaders()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.NO_CONTENT.value());
+
+        assertions.taskVariableWasUpdated(taskVariables.getProcessInstanceId(), "taskState", "assigned");
+        assertions.taskStateWasUpdatedInDatabase(taskId, "assigned", caseworkerCredentials.getHeaders());
+        assertions.taskFieldWasUpdatedInDatabase(taskId, "assignee",
+                                                 assigneeId, caseworkerCredentials.getHeaders());
+
+        result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            taskOperationRequest(TaskOperationName.EXECUTE_RECONFIGURE, OffsetDateTime.now().minus(Duration.ofDays(1))),
+            caseworkerCredentials.getHeaders()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.NO_CONTENT.value());
+
```

Comment:
can we also include checks on the verifiable task attributes that confirm the operation has completed successfully and as expected? I.e 
task state = assigned
reconfigure_request_time = null
last_reconfiguration_time = now()

Reply:
done